src/components/SelectInputContainer/index.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import Popup from '../Popup';
+import InputContainer, { InputContainerProps } from '../InputContainer';
+import RawInput from '../RawInput';
+import RawButton from '../RawButton';
+import List from '../List';
+import { useBlurEffect } from '../../hooks';
+
+import styles from './styles.css';
+
+type OptionKey = string | number;
+interface Option { [index: string]: any; }
+
+export interface SelectInputContainerProps<T extends OptionKey> extends Omit<InputContainerProps, 'input'> {
+    options: Option[];
+    optionKeySelector: (datum: Option, index: number) => T;
+    optionRenderer: React.ReactNode;
+    optionRendererParams: (optionKey: OptionKey, option: Option) => React.ReactNode;
+    onOptionClick: (optionKey: T) => void;
+    optionContainerClassName?: string;
+    onSearchInputChange: (search: string) => void;
+    valueDisplay: string;
+    persistantOptionPopup?: boolean;
+    searchPlaceholder?: string;
+    optionsPopupClassName?: string;
+    optionsPending?: boolean;
+    optionsEmptyComponent: React.ReactElement;
+}
+
+const GenericOptionRenderer = ({
```

Comment:
GenericOptionRenderer should have proper typings.
Also, prefer using `function GenericOptionRenderer() {...}` syntax